File: param_tutorial/src/calculate_client.cpp
#include "ros/ros.h"
#include "param_tutorial/Calculate.h"
#include <cstdlib>
 //atoi 함수 사용을 위한 라이브러리(문자열을 정수로 바꿔주는 함수)

int main(int argc, char** argv)
{
    ros::init(argc, argv, "calculate_client");
    
    if(argc != 3)
    {
        ROS_INFO("rosrun param_tutorial calculate_client a b");
        ROS_INFO("a,b : int32 number");
        return 1;
    }
    ros::NodeHandle nh;
    ros::ServiceClient calculate_client = nh.serviceClient<param_tutorial::Calculate>("calculate");

    param_tutorial::Calculate srv;
    srv.request.a = atoi(argv[1]);
    srv.request.b = atoi(argv[2]);

    if(calculate_client.call(srv)) 
    {
        ROS_INFO("a : %d, b : %d, result : %d",srv.request.a,srv.request.b,srv.response.result);

    }
    else
    {
        ROS_ERROR("Failed to call service");
        return 1;
    }
    return 0;
}  

File: param_tutorial/scripts/calculate_client.py
#!/usr/bin/python3
#-*- coding: utf-8 -*- 

import rospy
from param_tutorial.srv import Calculate, CalculateRequest
import sys

def calculate_client():
    rospy.init_node("calculate_client")

    if len(sys.argv) != 3:

        rospy.loginfo("rosrun param_tutorial calculate_client a b ")
        rospy.loginfo("a,b : int32 number")
        sys.exit(1)

    my_client = rospy.ServiceProxy("calculate",Calculate) #서비스 이름.서비스 타입(import한 것) 넣어줘야됨. 
    
    req = CalculateRequest()
    req.a = int(sys.argv[1])
    req.b = int(sys.argv[2])
            
    try:
        res = my_client(req)
        rospy.loginfo(f"a: {req.a}, b:{req.b}, result:{res.result}")

    except rospy.ServiceException as e:
            rospy.logerr(f"Failed : {e}")
            sys.exit(1)

if __name__ == "__main__":
    calculate_client()


        